source/code/plugins/scom_configuration.rb

```diff
@@ -59,11 +61,23 @@ def self.load_scom_configuration
         OMS::Log.error_once("Failed to read certificate or key from #{@@cert_path} #{@@key_path}")
         return false
       end # begin
+      
```

Comment:
This is really not what a certificate is meant to be used for. If the FQDN (or similar hostname subject) has changed since this certificate was created, then a new certificate should be generated.

If you are trying to actually retrieve the FQDN of the machine, please use the get_fully_qualified_domain_name method in [oms_common.rb](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/source/code/plugins/oms_common.rb).